presto-main/src/test/java/com/facebook/presto/sql/planner/TestLogicalPlanner.java

```diff
@@ -137,6 +137,20 @@ public void testInnerInequalityJoinWithEquiJoinConjuncts()
     }
 
     @Test
+    public void testLeftConvertedToInnerInequalityJoinNoEquiJoinConjuncts()
+            throws Exception
+    {
+        assertPlan("SELECT 1 FROM orders o LEFT JOIN lineitem l ON o.orderkey < l.orderkey WHERE l.orderkey IS NOT NULL",
+                anyTree(
+                        filter("O_ORDERKEY < L_ORDERKEY",
+                                join(INNER, ImmutableList.of(), Optional.empty(),
```

Comment:
I cannot do that in old matching framework. 
And as for the new one. Do we have mechanism for matching on expression shapes (i.e. translating symbols through projections)?